Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.mm

```diff
@@ -883,21 +883,22 @@ static bool isCopyDisplayedPixelBufferAvailable()
     provideMediaData(trackId);
 }
 
-void MediaPlayerPrivateWebM::appendCompleted()
+void MediaPlayerPrivateWebM::appendCompleted(bool success)
 {
     ASSERT(m_pendingAppends > 0);
     m_pendingAppends--;
     INFO_LOG(LOGIDENTIFIER, "pending appends = ", m_pendingAppends);
     setLoadingProgresssed(true);
     updateBufferedFromTrackBuffers(true);
+    m_errored |= !success;
```

Comment:
When we get an error, do we need to do anything here except waiting for load finished and m_pendingAppends getting to 0?
For instance, do we need to keep calling setLoadingProgresssed and updateBufferedFromTrackBuffers when m_errored is true?

Reply:
probably not,